api/src/clojure/org/akvo/flow_api/datastore/form_instance.clj

```diff
@@ -103,9 +104,14 @@
   [_ response-str]
   response-str)
 
+(def ^DateFormat sdf (SimpleDateFormat. "dd-MM-yyyy HH:mm:ss z"))
+
 (defmethod parse-response "DATE"
   [_ response-str]
-  (ds/to-iso-8601 (java.util.Date. (Long/parseLong response-str))))
+  (let [date (try (java.util.Date. (Long/parseLong response-str))
+                  (catch NumberFormatException e
+                    (.parse sdf response-str)))]
```

Comment:
this `parse` can also throw, right? https://docs.oracle.com/javase/8/docs/api/java/text/DateFormat.html#parse-java.lang.String-